### PR TITLE
Answer:14

### PR DESCRIPTION
**Pull request by reed756**
Added startWith operator

### CODE REVIEW

File: apps/rxjs-race-condition/src/app/app.component.ts [hunk 1]
```diff
@@ -19,12 +19,12 @@ export class AppComponent implements OnInit {
...
   dialog = inject(MatDialog);
   topicService = inject(TopicService);
-  topics: TopicType[] = [];
+  topics: TopicType[] | string[] = [];
 
   ngOnInit(): void {
     this.topicService
       .fakeGetHttpTopic()
-      .pipe(take(1))
+      .pipe(take(1), startWith(['Politic', 'Culture', 'Nature']))
```
Comment (tomalaforge) [inline apps/rxjs-race-condition/src/app/app.component.ts:27]: That's not the solution. Indeed you will get data inside your topics array, but not the right one. 

The goal is to see that imperative coding can create race condition. Refactor the code to declarative coding.

Comment (reed756) [reply apps/rxjs-race-condition/src/app/app.component.ts:27]: Thanks, I'll have another go

Comment (reed756) [reply apps/rxjs-race-condition/src/app/app.component.ts]: @tomalaforge Hey I've done a new commit which has a declarative solution now.

File: apps/rxjs-race-condition/src/app/topic.service.ts [hunk 2]
```diff
@@ -1,10 +1,29 @@
...
+      const data = {
+        topics: topics,
+      };
+      this.topics$.next(data);
+    });
+  }
+
+  getTopics() {
+    return this.topics$;
+  }
```
Comment (tomalaforge) [inline apps/rxjs-race-condition/src/app/topic.service.ts:28]: nit: prefer to return the observable and not the subject. 

The subject should not be accessible from the outside. 

Donc create a method, 

just do 
```ts
topics$ = this.topics$$.asObservable()
```

And use $$ or nothing for subject

File: apps/rxjs-race-condition/src/app/topic.service.ts [hunk 3]
```diff
@@ -1,10 +1,29 @@
...
+    this.init();
+  }
+
   fakeGetHttpTopic = () =>
     timer(1000).pipe(map((): TopicType[] => ['Politic', 'Culture', 'Nature']));
+
+  init() {
+    this.fakeGetHttpTopic().subscribe((topics) => {
+      const data = {
+        topics: topics,
```
Comment (tomalaforge) [inline apps/rxjs-race-condition/src/app/topic.service.ts:20]: nit, when the property and the value have the same name, you don't have to repeat it.

**Comment by tomalaforge** [comment]
This is way better. Nice job

**Comment by reed756** [comment]
Thank you for the review and the pointers to make it better, much appreciated!

**Pull request closed without merging**